contracts/toCKB-typescript/src/switch/atterm_redeem/mod.rs

```diff
@@ -1,7 +1,141 @@
 use crate::switch::ToCKBCellDataTuple;
-use crate::utils::types::Error;
+use crate::utils::config::SINCE_AT_TERM_REDEEM;
+use crate::utils::tools::{is_XT_typescript, XChainKind};
+use crate::utils::types::{Error, ToCKBCellDataView};
+use bech32::{self, FromBase32};
+use ckb_std::ckb_constants::Source;
+use ckb_std::error::SysError;
+use ckb_std::high_level::{load_cell_data, load_cell_lock_hash, load_cell_type, load_input_since};
 use core::result::Result;
+use molecule::prelude::*;
 
-pub fn verify(_toCKB_data_tuple: &ToCKBCellDataTuple) -> Result<(), Error> {
+fn verify_data(
+    input_toCKB_data: &ToCKBCellDataView,
+    out_toCKB_data: &ToCKBCellDataView,
+) -> Result<u128, Error> {
+    let lot_size = match input_toCKB_data.get_xchain_kind() {
+        XChainKind::Btc => {
+            if out_toCKB_data.get_btc_lot_size()? != input_toCKB_data.get_btc_lot_size()? {
+                return Err(Error::InvariantDataMutated);
+            }
+            let (hrp, data) = bech32::decode(
+                core::str::from_utf8(out_toCKB_data.x_unlock_address.as_ref()).unwrap(),
+            )
+            .map_err(|_| Error::XChainAddressInvalid)?;
+            if hrp != "bc" {
+                return Err(Error::XChainAddressInvalid);
+            }
+
+            let raw_data = Vec::<u8>::from_base32(&data).unwrap();
+            if raw_data.len() != 22 {
+                return Err(Error::XChainAddressInvalid);
+            }
+            if &raw_data[..2] != &[0x00, 0x14] {
+                return Err(Error::XChainAddressInvalid);
+            }
+
+            out_toCKB_data.get_btc_lot_size()?.get_sudt_amount()
+        }
+        XChainKind::Eth => {
+            if out_toCKB_data.get_eth_lot_size()? != input_toCKB_data.get_eth_lot_size()? {
+                return Err(Error::InvariantDataMutated);
+            }
+            if out_toCKB_data.x_unlock_address.as_ref().len() != 20 {
+                return Err(Error::XChainAddressInvalid);
+            }
+            out_toCKB_data.get_eth_lot_size()?.get_sudt_amount()
+        }
+    };
+    if input_toCKB_data.user_lockscript.as_ref() != out_toCKB_data.user_lockscript.as_ref()
+        || input_toCKB_data.x_lock_address.as_ref() != out_toCKB_data.x_lock_address.as_ref()
+        || input_toCKB_data.signer_lockscript.as_ref() != out_toCKB_data.signer_lockscript.as_ref()
```

Comment:
bump molecule version to 0.6.1
```suggestion
    if input_toCKB_data.user_lockscript != out_toCKB_data.user_lockscript
        || input_toCKB_data.x_lock_address != out_toCKB_data.x_lock_address
        || input_toCKB_data.signer_lockscript != out_toCKB_data.signer_lockscript
```